### PR TITLE
APP-304: Add semantic design variables

### PR DESCRIPTION
**Pull request by wgardiner**
## Description
This PR adds custom Tailwind variables based on our new design system. 

Each client has a `tailwind.css` (`tailwind.css` for Regen, and `Terrasos.tailwind.css` for Terrasos), which defines all the CSS variables, `tailwind.common.js` then references them and exposes them for useage in Tailwind generated class names. This allows for the same variable to be defined per client, and the appropriate version referenced in based on the active client. 

Closes: [APP-304](https://regennetwork.atlassian.net/browse/APP-304?atlOrigin=eyJpIjoiODE0NjhlODFkZDQxNGI3ZWE5ZTBlYmVhMDQ2Njg1NjEiLCJwIjoiaiJ9
)

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. Review Extend Colors in Storybook

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

[APP-304]: https://regennetwork.atlassian.net/browse/APP-304?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by wgardiner** [comment]
The variables were generated by using the [Design Tokens](https://www.figma.com/community/plugin/888356646278934516/design-tokens) plugin for Figma. I also evaluated Token Studio. I found some issues with consistency in references and wrote a script to preprocess the output then run it through a custom [Style Dictionary](https://amzn.github.io/style-dictionary/#/) script. There was some manual work to map the generated CSS variables and structure to the app and Tailwind, but ideally we'd have a workflow that automates this process so that we can easily keep design and our app in sync. In the future we might consider having a source of truth that both Figma and our app read from, rather than having to export from Figma.

**Comment by wgardiner** [comment]
I prefixed base colors, alias colors, and semantics with `bc`, `ac` and `s` respectively, just to prevent collisions during initial implementation, but we can easily update these based on feedback.

**Review by flagrede — COMMENTED** [review]
Nice work!

Should we commit the script that you used to build the CSS variables from the Figma export ?

Also I saw that MUI is not updated here, is this going to be handled in a follow-up PR?

**Comment by wgardiner** [comment]
I worked with Erika last week to get updated colors for `Terrasos.muiTheme.js` and `Terrasos.tailwind.css`. I also fixed and added scripts for generating theme file content from design tokens.

@flagrede are we already generating a storybook preview with the client .env variables set for Terrasos?

**Comment by flagrede** [comment]
@wgardiner I don't think so no.

File: design/.gitignore [hunk 1]
```diff
@@ -0,0 +1,176 @@
+build/
```
Comment (flagrede) [inline design/.gitignore:1]: Was there an issue with the root `.gitignore` to add a second one?

File: design/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- `semantic-regen-dark.css`: semantic colors for elements with a dark theme
+- `semantic-regen-light.css`: semantic colors for elements with a light theme
+- `semantic-terrasos-dark.css`: semantic colors for elements with a dark theme
+- `semantic-terrasos-light.css`: semantic colors for elements with a light theme
+- `tailwind.common.js`: a shared js file that can reference the relevant css variables for all theme clients
+
+These files can then be integrated into our app's theming system, by updating 
+- `tailwind.common.js`: shared interface
+- `tailwind.css`: default theme variables
+- `web-marketplace/src/clients/*/*.tailwind.css`: client specific theme variables
```
Comment (flagrede) [inline design/README.md:23]: Great doc!

**Review by flagrede — APPROVED** [review]
Looks good, nice work on this PR!

File: design/README.md (hunk 2 above)
Comment (wgardiner) [reply design/README.md:23]: Thanks!

**Pull request merged**